src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1303,6 +1304,97 @@ class CommentOnStatementSegment(BaseSegment):
     )
 
 
+@postgres_dialect.segment(replace=True)
+class CreateIndexStatementSegment(BaseSegment):
+    """A `CREATE INDEX` statement.
+
+    As specified in https://www.postgresql.org/docs/13/sql-createindex.html
+    """
+
+    type = "create_index_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        Ref.keyword("UNIQUE", optional=True),
+        Ref("OrReplaceGrammar", optional=True),
+        "INDEX",
+        Ref.keyword("CONCURRENTLY", optional=True),
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("IndexReferenceSegment", optional=True),
+        "ON",
+        Ref.keyword("ONLY", optional=True),
+        Ref("TableReferenceSegment"),
+        OneOf(
+            Sequence("USING", Ref("FunctionSegment"), optional=True),
+            Bracketed(
+                Delimited(
+                    Sequence(
+                        OneOf(
+                            Ref("ColumnReferenceSegment"),
+                            OptionallyBracketed(Ref("FunctionSegment")),
+                            Bracketed(Ref("ExpressionSegment")),
+                        ),
+                        AnyNumberOf(
+                            Sequence(
+                                "COLLATE",
+                                OneOf(
+                                    Ref("LiteralGrammar"),
+                                    Ref("QuotedIdentifierSegment"),
+                                ),
+                            ),
+                            Sequence(
+                                Ref("ParameterNameSegment"),
+                                Bracketed(
+                                    Delimited(
+                                        Sequence(
+                                            Ref("ParameterNameSegment"),
+                                            Ref("EqualsSegment"),
+                                            OneOf(
+                                                Ref("LiteralGrammar"),
+                                                Ref("QuotedIdentifierSegment"),
+                                            ),
+                                        ),
+                                        delimiter=Ref("CommaSegment"),
```

Comment:
Is this needed? Isn't comma the default?

Can see ANSI also used this but still don't think it's necessary - probably should clean them up.

BTW three instances of this in total in this PR.

Reply:
Actually looks like a lot more than three instances so let's leave this for now.